x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/index.ts

```diff
@@ -30,23 +23,26 @@ type QueuedPdfExecutorFactory = ExecuteJobFactory<ESQueueWorkerExecuteFn<JobDocP
 
 export const executeJobFactory: QueuedPdfExecutorFactory = async function executeJobFactoryFn(
   reporting: ReportingCore,
-  server: ServerFacade,
-  elasticsearch: ElasticsearchServiceSetup,
   parentLogger: Logger
 ) {
-  const browserDriverFactory = await reporting.getBrowserDriverFactory();
```

Comment:
This PR starts moving async things in execute job factory function to be resolved in the function that is returned instead of before, so the factory function can return immediately instead of returning a promise. Having `reporting.getConfig()` return synchronously ([instead of async](https://github.com/elastic/kibana/pull/55882)) kind of helps get that idea working.